packages/taiko-client/driver/preconf_blocks/server.go

```diff
@@ -298,6 +302,231 @@ func (s *PreconfBlockAPIServer) OnUnsafeL2Payload(
 	return nil
 }
 
+// OnUnsafeL2Response implements the p2p.GossipIn interface.
+func (s *PreconfBlockAPIServer) OnUnsafeL2Response(
```

Comment:
we can probably just reuse OnUnsafeL2Payload here and change some logs/checks....